src/java/org/apache/cassandra/io/sstable/format/DataComponent.java

```diff
@@ -86,12 +86,17 @@ private static CompressionParams buildCompressionParams(TableMetadata metadata,
                     compressionParams = CompressionParams.NOOP;
                     break;
                 case fast:
-                    if (!compressor.recommendedUses().contains(ICompressor.Uses.FAST_COMPRESSION))
+                    // The default compressor is generally Fast, but just in case we verify it.
+                    CompressionParams defaultCompressionParams = CompressionParams.defaultParams(metadata.keyspace);
+                    ICompressor maybeFastCompressor = defaultCompressionParams.getSstableCompressor();
+                    if (maybeFastCompressor.recommendedUses().contains(ICompressor.Uses.FAST_COMPRESSION)) {
```

Comment:
I would fix this logic. For the system keyspaces, we should not change the default compression at all as the defaults might change, regardless of whether it is "fast" for the system keyspaces or not.